pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -1321,6 +1321,12 @@ type ObjectStoreSpec struct {
 	// +optional
 	// +nullable
 	Security *SecuritySpec `json:"security,omitempty"`
+
+	// Type of ServerSideEncryption for RGW, if it is null then configure with kms
+	// +optional
+	// +nullable
+	// +kubebuilder:validation:Enum={"kms","s3","kms, s3"}
+	SSEType string `json:"ssetype,omitempty"`
```

Comment:
I am still wondering do we need to use separate `SecuritSpec` for `sse:s3`instead of this Enum. So the vault server and other backend details like the path for the kv engine need to be shared.For RGW both configurations are separated using different settings

Reply:
FYI @alimaredia @mattbenjamin @cbodley ^^

Reply:
@thotz I don't really know what the implications of that are, it sounds possible?

Reply:
I personally would expect this to be in `SecuritySpec` since the type holds all the encryption/security-related things.

Reply:
SecuritySpec applies both to clusters and object stores, so perhaps we need a new type. But let's separate the `ssetype` to multiple settings more more readability and flexibility. If another type is added in the future, the combinations will explode. For example, something like this?
```yaml
  serverSideEncryption:
    kmsEnabled: true
    s3Enabled: true
```